airflow/decorators/base.py

```diff
@@ -101,22 +102,22 @@ class DecoratedOperator(BaseOperator):
     :type kwargs_to_upstream: dict
     """
 
-    template_fields: Iterable[str] = ('op_args', 'op_kwargs')
+    template_fields: Sequence[str] = ('op_args', 'op_kwargs')
     template_fields_renderers = {"op_args": "py", "op_kwargs": "py"}
 
     # since we won't mutate the arguments, we should just do the shallow copy
     # there are some cases we can't deepcopy the objects (e.g protobuf).
-    shallow_copy_attrs = ('python_callable',)
+    shallow_copy_attrs: Tuple[str, ...] = ('python_callable',)
```

Comment:
```suggestion
    shallow_copy_attrs: Sequence[str] = ('python_callable',)
```

Reply:
The `Tuple` typing is consistent with `BaseOperator`. I can change both to `Sequence`. Otherwise MyPy cries if we just change `DecoratedOperator`:
```
airflow/decorators/base.py:110: error: Incompatible types in assignment (expression has type "Sequence[str]", base class "BaseOperator" defined the type as "Tuple[str, ...]")
        shallow_copy_attrs: Sequence[str] = ('python_callable',)
```

Reply:
We should change the type hint on `BaseOperator` to `Sequence`. It does not make sense to restrict it to `Tuple`.